internal/driver/config_test.go

```diff
@@ -51,3 +54,48 @@ func TestUpdateFromRaw(t *testing.T) {
 		})
 	}
 }
+
+// TestGetCameraXAddr verify the parsing of the camera XAddr
+func TestGetCameraXAddr(t *testing.T) {
+
+	tests := []struct {
+		input    map[string]models.ProtocolProperties
+		expected string
+
+		errorExpected bool
+	}{
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					Address: "http://localhost",
+					Port:    "80",
+				},
+			},
+			expected: "http://localhost:80",
```

Comment:
I think its good to add `errorExpected` field to all inputs, it can be added here.

Reply:
I think @ajcasagrande suggested we leave it off if it is `false`?